### PR TITLE
Add Cellpose v3.0.1

### PR DESCRIPTION
**Pull request by pavanvidem**
- Currently uses container as a requirement
- Added 6 more tests
- The output tiffs are completely black. No idea if that's normal. It is not a Galaxy issue. Standalone also generates black images.

### CODE REVIEW

File: tools/cellpose/cellpose.xml [hunk 1]
```diff
@@ -0,0 +1,154 @@
+<tool id="cellpose" name="Run Cellpose segmentation" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="23.2">
```
Comment (bgruening) [inline tools/cellpose/cellpose.xml:1]:
```suggestion
<tool id="cellpose" name="Run Cellpose segmentation" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="22.05">
```

File: tools/cellpose/cellpose.xml [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    <description>a generalist algorithm for cell and nucleus segmentation</description>
+    <macros>
+	<token name="@TOOL_VERSION@">3.0.1</token>
+	<token name="@VERSION_SUFFIX@">0</token>
+    </macros>
+    <requirements>
+	    <container type="docker">biocontainers/cellpose:3.0.1_cv1</container>
+	    <requirement type="package" version="0.22.0">scikit-image</requirement>
```
Comment (bgruening) [inline tools/cellpose/cellpose.xml:9]: either container of packages, not both ...

File: tools/cellpose/cellpose.xml [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    </requirements>
+    <stdio>
+        <exit_code range="1:" level="fatal" description="Error occurred. Please check Tool Standard Error"/>
+    </stdio>
+    <version_command>echo "@VERSION@"</version_command>
+    <command>
```
Comment (bgruening) [inline tools/cellpose/cellpose.xml:15]: maybe use `detect_errors=""` ...

File: tools/cellpose/cellpose.xml [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    </configfiles>
+    <inputs>
+        <param name="img_in" type="data" format="ome.tiff,tiff,jpeg,png" label="Choose the image file for segmention (usually after registration)"/>
+        <param name="model_type" type="select" label="Choose the pre-trained model type">
+            <option value="nuclei" selected="true">nuclei</option>
+            <option value="cyto">cyto</option>
+            <option value="cyto2">cyto2</option>
+            <option value="cyto3">cyto3</option>
+        </param>
+        <param argument="chan" value="0" type="integer" label="Type in the channel to segment" help="0=grayscale, 1=red, 2=green, 3=blue"/>
```
Comment (bgruening) [inline tools/cellpose/cellpose.xml:41]: why not a select here if there are only 3 options?

File: tools/cellpose/cellpose.xml [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+        <param argument="chan2" value="0" type="integer" optional="true" label="Type in the channel for nuclei segmatation (Optional)" help="0=None (will set to zero), 1=red, 2=green, 3=blue"/>
```
Comment (bgruening) [inline tools/cellpose/cellpose.xml:42]: same here

File: tools/cellpose/cellpose.xml [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+	    <param name="use_gpu" type="boolean" truevalue="booltrue" falsevalue="boolfalse" checked="false" label="Whether to use GPU?" />
+        <section name="options" title="Advanced Options" expanded="False">
+            <param argument="diameter" type="float" optional="true" label="Cell or nuclei diameter in pixels" help="Leave blank for automated estimation."/>
+            <param name="resample" type="boolean" truevalue="booltrue" falsevalue="boolfalse" checked="true" label="Run dynamics on the resampled image?"
+                help="Interpolated flows at the true image size. This option will create smoother ROIs when the cells are large but will be slower in case"/>
+            <param argument="flow_threshold" type="float" value="0.4" label="Flow error threshold (all cells with errors below threshold are kept) (not used for 3D)"/>
+            <param argument="cellprob_threshold" type="float" value="0.0" label="Cell probability threshold (all pixels with prob above threshold kept for masks)"/>
+            <param argument="niter" type="integer" value="0" label="Number of iterations"
```
Comment (bgruening) [inline tools/cellpose/cellpose.xml:52]: add min/max ? also to other params?

File: tools/cellpose/cellpose.xml [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+                title = {Cellpose: a generalist algorithm for cellular segmentation},
+                elocation-id = {2020.02.02.931238},
+                year = {2020},
+                doi = {10.1101/2020.02.02.931238},
+                publisher = {Cold Spring Harbor Laboratory},
+                URL = {https://www.biorxiv.org/content/early/2020/02/03/2020.02.02.931238},
+                eprint = {https://www.biorxiv.org/content/early/2020/02/03/2020.02.02.931238.full.pdf},
+                journal = {bioRxiv}
+            }
+        </citation>
```
Comment (bgruening) [inline tools/cellpose/cellpose.xml:152]:
```suggestion
        <citation type="doi">10.1101/2020.02.02.931238</citation>
```

File: tools/cellpose/cellpose.xml (hunk 2 above)
Comment (pavanvidem) [reply tools/cellpose/cellpose.xml]: removed the conda

File: tools/cellpose/cellpose.xml (hunk 4 above)
Comment (pavanvidem) [reply tools/cellpose/cellpose.xml]: yes, makes sense

File: tools/cellpose/cellpose.xml (hunk 6 above)
Comment (pavanvidem) [reply tools/cellpose/cellpose.xml]: It is hard to find min/max for all the params. Tried to add as much as I could.

**Pull request merged**